### ISSUE TITLE
[BUG] GPU training causes invalid OpenCL state

### ISSUE THREAD

**Issue by xEcho1337** [brain4j-org/brain4j#30]
## Bug Description
Using the GPU for medium sized training tasks causes invalid OpenCL state (CL_OUT_OF_RESOURCES).

## Steps to Reproduce
1. Create a model
2. Use the MNIST dataset
3. Start the training using the GPU

## Expected Behavior
Loss should decrease over time and the GPU should speed up training.

## Actual Behavior
Brain4J crashes after few computed batches.
What actually happened (including any error messages, stack traces, or console output).

## Environment
- brain4j version: 3.0.0
- Java version: 21
- Operating system: Windows

## Additional Context
MLP with 784 -> 128 (ReLU) -> 64 (ReLU) -> 10 (Softmax), MNIST dataset, 128 batch size.

**Comment by Adversing** [comment]
NOTE: This issue is currently related to NVIDIA GPU ecosystem only.

Tested devices: RTX 4070S, RTX 5070.

**Comment by xEcho1337** [comment]
> NOTE: This issue is currently related to NVIDIA GPU ecosystem only.
> 
> Tested devices: RTX 4070S, RTX 5070.

I can confirm, it works fine (almost, there's still a memory leak) on my MacBook Air M3, this makes the issue even stranger.

I'm starting to suspect it's a bug in Windows's OpenCL driver.

**Comment by xEcho1337** [comment]
After some tests, looks like there was a memory leak. It should be solved, even though now crashes with another code: CL_INVALID_KERNEL_ARGS.
